inc/Engine/Media/ImageDimensions/ImageDimensions.php

```diff
@@ -227,7 +226,15 @@ private function get_local_path( $url ) {
 			return $this->local_paths[ md5( $url ) ];
 		}
 
-		return str_replace( content_url(), rocket_get_constant( 'WP_CONTENT_DIR' ), $url );
+		if ( false === strpos( $url, content_url() ) ) {
+			$url = site_url( '/' ) . ltrim( $url, '/' );
+		}
+
+		return str_replace(
+			'/',
+			rocket_get_constant( 'DIRECTORY_SEPARATOR' ),
+			str_replace( site_url(), rocket_get_constant( 'ABSPATH' ), $url )
```

Comment:
I'm thinking that this might not be working on some WP setups, notably when WP is installed in a sub-directory like so:
- home URL is http://example.org
- site URL is http://example.org/wp

Can you test it?

Reply:
I adjusted the tests to accept changing `site_url` and `home_url` and tests passed.

You can check this fixture
https://github.com/wp-media/wp-rocket/blob/4035cff4b1ef03f07055022202a162f923e06ef8/tests/Fixtures/inc/Engine/Media/ImageDimensions/ImageDimensions/specifyImageDimensions.php#L213-L224